fabtests/multinode/src/core.c

```diff
@@ -311,60 +415,55 @@ int multinode_run_test()
 			if (ret)
 				return ret;
 
-			ret = method.send();
+			ret = method.recv();
 			if (ret)
 				return ret;
 
-			ret = method.send();
+			ret = method.wait();
 			if (ret)
 				return ret;
-
-			pm_barrier();
 		}
+		ret = send_recv_barrier();
+		if (ret)
+			return ret;
```

Comment:
I don't understand the changes in this function.  Maybe an earlier patch is wrong?